gtsam/linear/NoiseModel.cpp

```diff
@@ -136,14 +136,18 @@ bool Gaussian::equals(const Base& expected, double tol) const {
   const Gaussian* p = dynamic_cast<const Gaussian*> (&expected);
   if (p == NULL) return false;
   if (typeid(*this) != typeid(*p)) return false;
-  //if (!sqrt_information_) return true; // ALEX todo;
+  //TODO(Alex);
+  //if (!sqrt_information_) return true;
   return equal_with_abs_tol(R(), p->R(), sqrt(tol));
 }
 
 /* ************************************************************************* */
 Vector Gaussian::sigmas() const {
-  // TODO(frank): can this be done faster?
-  return Vector((thisR().transpose() * thisR()).inverse().diagonal()).cwiseSqrt();
+  Matrix R = thisR();
```

Comment:
Explain why this is faster?

Reply:
Since we know R is upper triangular, using `triangularView` lets us use `.solve(m)` which effectively performs `this.inverse()*m` but in an efficient way using backsubstitution with the triangular matrix. Thus `n.solve(I) = n.inverse()*I = n.inverse()` which is cheaper than a full inverse aka `< O(n^3)`.

Reply:
https://eigen.tuxfamily.org/dox/classEigen_1_1TriangularViewImpl_3_01__MatrixType_00_01__Mode_00_01Dense_01_4.html#a911664ccf5522c778ffc5405247e8a59

Reply:
Actually, you are right, but in that case we should do that saving in `covariance`, moved to the .cpp file. And here just say `covariance(). diagonal()).cwiseSqrt()`

Reply:
Done. Running tests locally and then will commit.